Support number, boolean and null as AppOption values

ivarne opened the pull request:
Add support for multiple types as AppOptionValues. The internal representation is still string for simplicity for app developers and backwards compatibility, but you can specify a type for json serialisation.


## Related Issue(s)
- https://github.com/Altinn/app-lib-dotnet/issues/413

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

ivarne committed:
Support number, boolean and null as AppOption values

olemartinorg commented:
Removing myself as reviewer on this one. I don't touch C# enough to be any good at reviewing the actual code, but I'll gladly test the solution at a later stage if needed. :blush:

ivarne committed:
Add tests, json attributes and serialize desctiption

ivarne commented:
Added tests and fixed a bug in not serialising description. Added @tjololo as reviewer instead, but the most important verification is that @olemartinorg thinks it solves the frontend issue. Backend doesn't yet have any direct usage of app options (apart from letting them through)

tjololo reviewed (commented):
Have a look at the CodeQL and Sonar warnings. Mainly the nullability and complexity issues

ivarne committed:
Fix potential nullability issue and improve test coverage

ivarne committed:
Fix tolower typo

tjololo reviewed (approved):
:godmode:
